### ISSUE TITLE
Unable to import Support Files 

### ISSUE THREAD

**Issue by Symplicityy** [osy/Jitterbug#1]
When trying to import the Support Files, they are grayed out as if they are an unsupported filetype.
![0559A377-1D76-492B-B5F2-98D46F9568BA](https://user-images.githubusercontent.com/17555839/120407358-ff9d5880-c31a-11eb-8fda-79beca5a727b.png)

**Comment by osy** [comment]
Is this installed with AltStore? I think there’s a similar issue on UTM as well. Can you try using https://dantheman827.github.io/ios-app-signer/

**Comment by Symplicityy** [comment]
I am still unable to select the files after installing with iOS app signer.

**Comment by Symplicityy** [comment]
I was able to get the files to appear by using Filza to create a folder named “SupportImages” in the documents folder and copying the files into there. This works for me as a temporary workaround. Assuming iTunes files sharing is enabled, this could also be done on a non-jailbroken device.

**Comment by osy** [comment]
Did you use the selector for importing the pairing as well? Or did you open the file? If second can you try the selector and see if there’s the same issue?

**Comment by Symplicityy** [comment]
I tried that before when I installed the app with iOS app signer. It wasn't grayed out like the other files but when I clicked it to import it nothing happened so I did it the other way instead.

**Comment by BluDood** [comment]
Same thing happens for me, but the files I'm trying to import aren't grayed out.
![IMG_9B11016A32B2-1](https://user-images.githubusercontent.com/70075053/120513296-6a5a9c80-c3cc-11eb-93a3-6b1cbcf57e95.jpeg)

**Comment by osy** [comment]
@BluDood what happens if you try to select one?

**Comment by BluDood** [comment]
> @BluDood what happens if you try to select one?

Nothing happens. Here is a screen recording of what happens when I tryhttps://user-images.githubusercontent.com/70075053/120517468-c6bfbb00-c3d0-11eb-98e6-78e0af9067d6.MOV

**Comment by Monkiey** [comment]
> @BluDood what happens if you try to select one?

Um sorry, different issue when I run the jitterbugpair script it crates the device pairing file so where is it stored on my mac so I can find it and transfer it to my ios device.

**Comment by Monkiey** [comment]
Nvm mind found it its located in a  user directory folder

**Comment by eddiestuczynski** [comment]
I just wanted to add that I am having the same problems with the support files being grayed out on Altstore installation! I am about to try the App Signer method to see if that will work! I am not going to bother supplying a screenshot because you already know the problems! hahaha

**Comment by eddiestuczynski** [comment]
> I just wanted to add that I am having the same problems with the support files being grayed out on Altstore installation! I am about to try the App Signer method to see if that will work! I am not going to bother supplying a screenshot because you already know the problems! hahaha

Plus everything happened the same ways as already reported! :)

**Comment by osy** [comment]
Hmm I am not able to reproduce both on jailbroken iOS 14.3 as well as AltStore signed 14.5.1. What os are people running?

**Comment by BluDood** [comment]
> Hmm I am not able to reproduce both on jailbroken iOS 14.3 as well as AltStore signed 14.5.1. What os are people running?

I am currently on IOS 14.3 unjailbroken

**Comment by Symplicityy** [comment]
I am almost positive that I just figured out what is causing the issue and a super easy way to get it working till there's a fix. I copied the 2 files from "iCloud Drive" to "On My Phone" and I was able to import them that way. It must be some kind of a bug with importing certain files from iCloud Drive because the pairing file imports fine through iCloud. Most people probably don't even realize the files are in iCloud because iOS downloads files from Safari into iCloud by default.

**Comment by BluDood** [comment]
> I am almost positive that I just figured out what is causing the issue and a super easy way to get it working till there's a fix. I copied the 2 files from "iCloud Drive" to "On My Phone" and I was able to import them that way. It must be some kind of a bug with importing certain files from iCloud Drive because the pairing file imports fine through iCloud. Most people probably don't even realize the files are in iCloud because iOS downloads files from Safari into iCloud by default.

Mine are in On my iPhone, not working

**Comment by BluDood** [comment]
> Very weird. I tried putting files into my iCloud and it still works for me. I can't fix this issue until I am able to reproduce it...

If I put the files in iCloud, they just get grayed out. Very weird indeed

**Comment by iMonZ** [comment]
All right I fixed it with installing it over Appsync unified and not IOS app signer. So its a signing issue

**Comment by jankais3r** [comment]
It works just fine for me, signed with iOS App Signer and my dev profile.

**Comment by ViRb3** [comment]
This looks like a signing issue. I have seen this happen if the provisioning profile's app id doesn't match the app's bundle id. If using an explicit provisioning profile, such as `com.test.123`, make sure to change the app's bundle id to the same value, instead of the default `com.osy86.Jitterbug`.

**Comment by iMonZ** [comment]
Quick workaround: it’s possible to add an option to share the files with the share menu

**Comment by sufilm** [comment]
I just have the same problem and I was installed with Abox but not AIstore. However, I just  reset my iPad and follow the same step, it just work and have no other problem.

**Comment by Symplicityy** [comment]
I just want to add that I also recently tried this on my M1 iPad Pro on iOS 14.5 (obviously unjailbroken) and I had the same issue. Copying the files from iCloud to my iPad fixed the issue once again.

**Comment by osy** [comment]
I think it's been established that the bug is caused by a signing issue with provisioning profiles's app id not matching the bundle id.

**Comment by ty-yqs** [comment]
Did you select the files from iCloud? On my devices, I can select the files from "My iPad/iPhone"

**Comment by Symplicityy** [comment]
> Did you select the files from iCloud? On my devices, I can select the files from "My iPad/iPhone"

No. I was never able to select the files from iCloud. I was only able to select them after copying them to my iPhone/iPad. Apparently installing the TestFlight version should allow you to select it from your iCloud though.

**Comment by ty-yqs** [comment]
Did you change Bundle ID when you signing it? Maybe because of the Bundle ID.

**Comment by osy** [comment]
Not sure if AltStore fixed the bug yet. You have to manually sign. Follow the readme.

**Comment by osy** [comment]
Copying manually always works. Like I said you have to follow the readme and sign with Xcode. Idk what ios-signer-service is but it probably has the same bug as AltStore.

**Comment by MrSco** [comment]
well i finally got it working... very cool stuff... i notice however, i have to be connected to a wifi ap when using the same device to debug and run app with vpn... is that just a limitation of the vpn tunnel? it needs the connection for the tunnel to run over? or am i doing it wrong?

vpn works great when i'm connected to a wifi ap though! thanks!

**Comment by ViRb3** [comment]
@MrSco I realized what the problem is. If you look at the bottom of the JitterBug README, you will actually see your exact issue: https://github.com/osy/Jitterbug#launch-fails-with-failed-to-get-the-task-for-process-xxx The description states that this happens when you are trying to use JitterBug to launch an app (like DolphiniOS) which itself does not have `get-task-allow` (debugging) entitlement. This means that you need to sign DolphiniOS with debugging entitlement, but not JitterBug itself.

**Comment by MrSco** [comment]
> @MrSco I realized what the problem is. If you look at the bottom of the JitterBug README, you will actually see your exact issue: https://github.com/osy/Jitterbug#launch-fails-with-failed-to-get-the-task-for-process-xxx The description states that this happens when you are trying to use JitterBug to launch an app (like DolphiniOS) which itself does not have `get-task-allow` (debugging) entitlement. This means that you need to sign DolphiniOS with debugging entitlement, but not JitterBug itself.

yes, that's what i was referring too.... UTM and Dolphin ipa's need to have debugger permissions enabled when uploaded/sign

**Comment by Kickflipkid687** [comment]
I was finally able to get it working as well.  I tried to many other methods, but only thing that worked, was to use that Github/IOS Signer method.  

I'm on Windows as well, so I wasn't sure how to get/setup the required? .p12 file, or I didn't use that at first/wrong file maybe and it didn't work.

So I had to go into my Apple Dev account, and request a certificate.  Download that. Then I found a tutorial on Youtube, on how to create the .p12 locally, with OpenSSL, and add a password to the .p12 file.   

https://youtu.be/0iLtDb2ZKAE

Then I was able to get that Raw Data as the tutorial says/do the rest, and it worked!  I didn't enable the Debug option like others said.   I just uploaded the Jitterbug.ipa/not Lite.  Then I think said to use the existing bundle ID/name maybe even.  

As someone said, i think you DO maybe need to have WIFI/Cell signal to run the VPN at first/open an app.   But then, you can turn OFF everything networking/go into airplane mode even, and it still works it seems like.  If you leave Jitterbug open.  

Sometimes it freaks out and says can't find files or something, but I just close Jitterbug/other app i'm trying to open with it, and re-launch Jitter, then it works.

**Comment by Spidy123222** [comment]
> @MrSco I realized what the problem is. If you look at the bottom of the JitterBug README, you will actually see your exact issue: https://github.com/osy/Jitterbug#launch-fails-with-failed-to-get-the-task-for-process-xxx The description states that this happens when you are trying to use JitterBug to launch an app (like DolphiniOS) which itself does not have `get-task-allow` (debugging) entitlement. This means that you need to sign DolphiniOS with debugging entitlement, but not JitterBug itself.

DolphiniOS uses get-task-allow they used it for a while since iOS 14.2 from what I know and when using AltStore or another signing program it still works.
